Tests/SonarScanner.MSBuild.PreProcessor.Test/AnalysisConfigGeneratorTests.cs

```diff
@@ -384,6 +384,77 @@ public void GenerateFile_ExcludeCoverage_Exclusions_MultipleSpecified()
             .Which.Value.Should().Be("coverage1.xml,coverage2.xml,coverage3.xml,coverage4.xml");
     }
 
+    [DataTestMethod]
+    [DataRow("coverage.xml", "", "", "", "coverage.xml", "")]
+    [DataRow("coverage.xml", "", "local.cs,local.js", "", "local.cs,local.js,coverage.xml", "")]
+    [DataRow("coverage.xml", "", "", "server.cs,server.js", "server.cs,server.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("coverage.xml", "", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("", "", "", "", "", "")]
+    [DataRow("", "", "local.cs,local.js", "", "local.cs,local.js", "")]
+    [DataRow("", "", "", "server.cs,server.js", "", "server.cs,server.js")]
+    [DataRow("", "", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js", "server.cs,server.js")]
+    [DataRow("", "coverage.xml", "", "", "coverage.xml", "")]
+    [DataRow("", "coverage.xml", "local.cs,local.js", "", "local.cs,local.js,coverage.xml", "")]
+    [DataRow("", "coverage.xml", "", "server.cs,server.js", "server.cs,server.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("", "coverage.xml", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js,coverage.xml", "server.cs,server.js")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "", "", "localCoverage.xml", "")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "local.cs,local.js", "", "local.cs,local.js,localCoverage.xml", "")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "", "server.cs,server.js", "server.cs,server.js,localCoverage.xml", "server.cs,server.js")]
+    [DataRow("localCoverage.xml", "serverCoverage.xml", "local.cs,local.js", "server.cs,server.js", "local.cs,local.js,localCoverage.xml", "server.cs,server.js")]
```

Comment:
I did some testing locally and now it works fine.
Please, @gregory-paidis-sonarsource, review carefully all the behavior tested here; I would suggest also doing some local analysis on your side to confirm the proper behavior of the scanner.